### PR TITLE
Add expression for square by diagonal and rectangle from 3 points

### PR DESCRIPTION
**Pull request by lbartoletti**
## Description
With the new class QgsQuadrilateral, we can add expressions to create a square by a diagonal and rectangles by 3 points

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/core/expression/qgsexpressionfunction.cpp [hunk 1]
```diff
@@ -2487,6 +2488,73 @@ static QVariant fcnMakeRegularPolygon( const QVariantList &values, const QgsExpr
 
 }
 
+static QVariant fcnMakeSquare( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QgsGeometry pt1 = QgsExpressionUtils::getGeometry( values.at( 0 ), parent );
+  if ( pt1.isNull() )
+  {
+    qDebug() << " pt1 null\n";
```
Comment (nyalldawson) [inline src/core/expression/qgsexpressionfunction.cpp]: Remove debug messages

File: src/core/expression/qgsexpressionfunction.cpp [hunk 2]
```diff
@@ -4888,8 +4956,17 @@ const QList<QgsExpressionFunction *> &QgsExpression::Functions()
                                             << QgsExpressionFunction::Parameter( QStringLiteral( "geometry" ) )
                                             << QgsExpressionFunction::Parameter( QStringLiteral( "number_sides" ) )
                                             << QgsExpressionFunction::Parameter( QStringLiteral( "circle" ), true, 0 ),
-                                            fcnMakeRegularPolygon, QStringLiteral( "GeometryGroup" ) );
-
+                                            fcnMakeRegularPolygon, QStringLiteral( "GeometryGroup" ) )
+        << new QgsStaticExpressionFunction( QStringLiteral( "make_square" ), QgsExpressionFunction::ParameterList()
+                                            << QgsExpressionFunction::Parameter( QStringLiteral( "geometry" ) )
```
Comment (nyalldawson) [inline src/core/expression/qgsexpressionfunction.cpp]: Argument names should be distinct and match those from the function documentation., Eg. "point1"

**Review by nyalldawson — COMMENTED** [review]
Looking good!

**Comment by nyalldawson** [comment]
Also don't forget the "[feature]" tag in the commit message

File: src/core/expression/qgsexpressionfunction.cpp (hunk 1 above)
Comment (lbartoletti) [reply src/core/expression/qgsexpressionfunction.cpp]: Oops, done. Thanks :)

**Comment by m-kuhn** [comment]
@lbartoletti `[feature]` should be on the commit ( not the PR title )

**Comment by pigreco** [comment]
example: the function used is not the correct one
geom_to_wkt(`make_rectangle`(make_point(0, 0), make_point(0,5), make_point(5, 5), 0))) → 'Polygon ((0 0, 0 5, 5 5, 5 0, 0 0))'

**Pull request merged**